bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -172,9 +173,33 @@ public boolean tableExists(String tableName) throws IOException {
     return response;
   }
 
+  // Workaround for #2016, BigtableTableAdminClient#getTable throws TableNotFoundException(due to
+  // two step details fetch) and an intermediate exception(FAILED_PRECONDITION).
+  //
+  // This method is only being called when user sends `null` or `""` string to listTable(Pattern)
+  // which in being used by deleteTables(), enableTables(), disableTables(),
+  // getTableDescriptorByTableNames().
```

Comment:
I dont think explanation is quite right, I think this would be more appropriate:

```
BigtableTableAdmin#listTables doesn't include table details (like column families), so the table descriptors must be fetched into 2 phases:
1. list all of the names 
2. fetch the table details

Due to the non-atomic listing, tables in the list might disappear before fetching the details. This method will suppress those tables
```